.github/workflows/update_app_templates_hash.yml

```diff
@@ -0,0 +1,43 @@
+name: Update aws/aws-sam-cli with latest commit hash from aws/aws-sam-cli-app-templates
+
+on:
+  schedule:
+    - cron: "0 * * * *"
+  workflow_dispatch: {}
+
+jobs:
+  updateCommitHash:
+    permissions:
+      pull-requests: write
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout App Templates
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli-app-templates
+          path: aws-sam-cli-app-templates
+
+      - name: Checkout SAM CLI
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli
+          path: aws-sam-cli
+
+      - name: Update commit hash in SAM CLI from App Templates
+        run: |
+          cd aws-sam-cli-app-templates
+          export APP_TEMPLATES_COMMIT_HASH=$(git rev-parse HEAD)
+          cd ../aws-sam-cli
+          cat <<< $(jq --arg commit_hash "$APP_TEMPLATES_COMMIT_HASH" --indent 2 '.app_template_repo_commit =  $commit_hash' samcli/runtime_config.json) > samcli/runtime_config.json
+
+      - name: Raise PR for SAM CLI
+        uses: peter-evans/create-pull-request@v4
```

Comment:
I'd be wary of using third-party actions without reviewing the code and pinning to a hash. If that action introduces buggy/malicious code and they update the tag `v4` to point to that commit, it'll run the code in our repository, which can be an attack vector.

I'd much rather use the official GitHub CLI which is already bundled in GitHub-hosted runners. We can use [`gh pr create`](https://cli.github.com/manual/gh_pr_create).

Something like this (untested):

```yaml
      - run: gh pr create [...]
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Thanks for the suggestion! I've already have alternative way to create PR through GH CLI. Let me test with that one and update the workflow definition.